src/Disks/ObjectStorages/S3/registerDiskS3.cpp

```diff
@@ -65,7 +65,34 @@ void checkReadAccess(const String & disk_name, IDisk & disk)
         throw Exception("No read access to S3 bucket in disk " + disk_name, ErrorCodes::PATH_ACCESS_DENIED);
 }
 
-void checkRemoveAccess(IDisk & disk) { disk.removeFile("test_acl"); }
+void checkRemoveAccess(IDisk & disk)
+{
+    disk.removeFile("test_acl");
+}
+
+bool checkBatchRemoveCapability(S3ObjectStorage & storage, const String & key)
+{
+    String path = key + "/test_acl";
+    auto file = storage.writeObject(path, WriteMode::Rewrite);
+    try
+    {
+        file->write("test", 4);
+    }
+    catch (...)
+    {
+        file->finalize();
+        return true; /// We don't have write access, therefore no information about batch remove.
```

Comment:
Hm, should we not return "false" if we don't have information about batch remove?